### ISSUE TITLE
Cox regression

### ISSUE THREAD

**Issue by jeffwong** [dmlc/xgboost#326]
Would be great to be able to do cox regression like in the gbm R package. This would require supporting the surv object for censoring, and also having a function that returns the base hazard.

I'll try to develop a PR for this, but that will probably take me a long time. Would be great if someone can read through the gbm package and port this over

**Comment by tqchen** [comment]
You can first try to make a version with the customized loss function from
the R side. Which might be easier

**Comment by tqchen** [comment]
I am marking this as a wishlist, to see if anyone is interested in contributing this

**Comment by tqchen** [comment]
https://github.com/dmlc/xgboost/issues/749
